### ISSUE TITLE
Inserting Decimals Results in pyodbc.DataError: ('String data, right truncation: length 22 buffer 20', '22001') for Some Decimals

### ISSUE THREAD

**Issue by isosphere** [mkleehammer/pyodbc#367]
I'm back again to punish myself by trying to use decimal.Decimal with MSSQL! Previously I had an issue where decimal.Decimal(float(number)) required > 34 digits of precision, but I think that's irrelevant here.

**System**

- Windows 10, x64
- Python 3.6.4 x64
-- pyodbc 4.0.22
- MSSQL 13.0.4001

**Table**
```
CREATE TABLE [dbo].[examplemodel](
	[id] [int] IDENTITY(1,1) NOT NULL,
	[date] [date] NOT NULL,
	[reference_date] [date] NOT NULL,
	[value] [numeric](**10**, 4) NULL,
	[updated] [datetime2](7) NOT NULL,
PRIMARY KEY CLUSTERED 
(
	[id] ASC
)WITH (PAD_INDEX = OFF, STATISTICS_NORECOMPUTE = OFF, IGNORE_DUP_KEY = OFF, ALLOW_ROW_LOCKS = ON, ALLOW_PAGE_LOCKS = ON) ON [PRIMARY],
 CONSTRAINT [examplemodel_date_reference_date_56ac3056_uniq] UNIQUE NONCLUSTERED 
(
	[date] ASC,
	[reference_date] ASC
)WITH (PAD_INDEX = OFF, STATISTICS_NORECOMPUTE = OFF, IGNORE_DUP_KEY = OFF, ALLOW_ROW_LOCKS = ON, ALLOW_PAGE_LOCKS = ON) ON [PRIMARY]
) ON [PRIMARY]
GO
```

**Inserting a Row**
```
import pyodbc
import datetime

connection = pyodbc.connect(
    r'DRIVER={SQL Server Native Client 11.0};'
    r'SERVER=myhostname\SQLExpress;'
    r'DATABASE=databasename'
    r'Trusted_Connection=yes;'

)

cursor = connection.cursor()

cursor.execute(
    'SET NOCOUNT ON INSERT INTO [examplemodel] ([date], [reference_date], [value], [updated]) VALUES (?, ?, ?, ?); SELECT CAST(SCOPE_IDENTITY() AS bigint)',
    ('2022-02-01', '2019-02-01', '100000.5000', datetime.datetime(2017, 3, 1, 20, 30))
)
```
**pyodbc.DataError: ('String data, right truncation: length 22 buffer 20', '22001')**

The result can be changed by exclusively manipulating "value". Here's a list of values for value that cause the above error:

- '100000.0000' (10 digits)
- '150000.0000'
- '999999.1000'

These values do not cause an error:
- '100000.00' (9 digits)
- '100000.000'
- '10000.0000'
- '999999'
- '999999.1'
- '999999.100'

So from the above it appears my sin is using 10 digits. If I insert any of the problem values directly in SQL Server Management Studio, it doesn't give me any trouble. The string truncation error here is very confusing. 

I bugged the django-pyodbc-azure project about this first, but was able to recreate it exclusively with pyodbc. Here's my original post:
https://github.com/michiya/django-pyodbc-azure/issues/154

**Comment by isosphere** [comment]
Ah-ha! It looks like this is the same regression issue mentioned in #334. Downgrading to pyodbc 4.0.21 resolves the issue for me.

**Comment by mkleehammer** [comment]
This has been fixed in 4.0.23, which reverts the broken code in 4.0.22.  Thanks.
